app/models/solr_document.rb

```diff
@@ -82,7 +82,7 @@ def based_near_linked
   end
 
   def based_near_linked_label
-    ScholarsArchive::LabelParserService.parse_label_uris(self['based_near_linked_ssim'])
+    ScholarsArchive::LocationLabelParserService.location_parse_uris(ScholarsArchive::LabelParserService.parse_label_uris(self['based_near_linked_ssim']))
```

Comment:
If ScholarsArchive::LabelParseService needs more functionality, id rather that exist in the ::LabelParseService rather than create a new object that does the work. Unless im missing something about how ::LabelParseService needs to still function like it is, but in this one instance needs to be extended.